src/Symfony/Component/Routing/Matcher/Dumper/CompiledUrlMatcherDumper.php

```diff
@@ -398,7 +398,7 @@ private function compileStaticPrefixCollection(StaticPrefixCollection $tree, \st
             }
 
             $state->mark += 3 + $state->markTail + \strlen($regex) - $prefixLen;
-            $state->markTail = 2 + \strlen($state->mark);
+            $state->markTail = 2 + \strlen((string) $state->mark);
```

Comment:
This seems to be not related to your PR. Did you encountered an issue without explicit cast?